R/merge.R

```diff
@@ -50,11 +50,30 @@ merge.data.table = function(x, y, by = NULL, by.x = NULL, by.y = NULL, all = FAL
       by = intersect(nm_x, nm_y)
     if (length(by) == 0L || !is.character(by))
       stopf("A non-empty vector of column names for `by` is required.")
-    if (!all(by %chin% intersect(nm_x, nm_y)))
-      stopf("Elements listed in `by` must be valid column names in x and y")
-    by = unname(by)
-    by.x = by.y = by
+    # UPDATED PART STARTS HERE
+  if (!all(by %chin% intersect(nm_x, nm_y))) {
+    # Identify which keys are missing from each data table
+    missing_x = setdiff(by, nm_x)
+    missing_y = setdiff(by, nm_y)
+    
```

Comment:
@venom1204, do you see a "check warning" from the "lint-r" check around this line when you visit the [files changed](https://github.com/Rdatatable/data.table/pull/6691/files) tab of the pull request? It is asking you to remove the spaces on line 58. Since it doesn't contain any text, it shouldn't contain any spaces either.